.devcontainer/Dockerfile

```diff
@@ -68,8 +68,75 @@ ENV DEBIAN_FRONTEND=noninteractive
+RUN add-apt-repository -y 'deb https://cran.rstudio.com/bin/linux/ubuntu focal-cran40/'
+RUN apt update
+RUN apt-get -y install r-base
+
+# copy common dependency installation scripts
+RUN mkdir -p /opt/rstudio-tools/dependencies/linux
+RUN mkdir -p /opt/rstudio-tools/dependencies/tools
+RUN mkdir -p /opt/rstudio-tools/dependencies/common
+COPY ./dependencies/tools/rstudio-tools.sh /opt/rstudio-tools/dependencies/tools/
+
+# copy panmirror project setup (so install-common can install needed deps)
+RUN mkdir -p /opt/rstudio-tools/panmirror
+COPY ./src/gwt/panmirror/src/editor/yarn.lock /opt/rstudio-tools/panmirror/
+COPY ./src/gwt/panmirror/src/editor/package.json /opt/rstudio-tools/panmirror/
+
+# install common dependencies
+# here, we deviate from the typical "./install-common" script and instead
+# call the individual dependency scripts for better layer caching.
+
+# crashpad
+COPY ./dependencies/common/install-crashpad /opt/rstudio-tools/dependencies/common
+RUN cd /opt/rstudio-tools/dependencies/common && /bin/bash ./install-crashpad bionic
+
+# dictionaries
+COPY ./dependencies/common/install-dictionaries /opt/rstudio-tools/dependencies/common/
+RUN cd /opt/rstudio-tools/dependencies/common && /bin/bash ./install-dictionaries
+
+# mathjax
+COPY ./dependencies/common/install-mathjax /opt/rstudio-tools/dependencies/common/
+RUN cd /opt/rstudio-tools/dependencies/common && /bin/bash ./install-mathjax
+
+# boost
+COPY ./dependencies/common/install-boost /opt/rstudio-tools/dependencies/common/
+RUN cd /opt/rstudio-tools/dependencies/common && /bin/bash ./install-boost
+
+# pandoc
+COPY ./dependencies/common/install-pandoc /opt/rstudio-tools/dependencies/common/
+RUN cd /opt/rstudio-tools/dependencies/common && /bin/bash ./install-pandoc
+
+# R packages
+COPY ./dependencies/common/install-packages /opt/rstudio-tools/dependencies/common/
+RUN cd /opt/rstudio-tools/dependencies/common && /bin/bash ./install-packages
+
+# sentry
+COPY ./dependencies/common/install-sentry-cli /opt/rstudio-tools/dependencies/common/
+RUN cd /opt/rstudio-tools/dependencies/common && /bin/bash ./install-sentry-cli
+
+# NPM dependencies
+COPY ./dependencies/common/install-npm-dependencies /opt/rstudio-tools/dependencies/common/
+RUN cd /opt/rstudio-tools/dependencies/common && /bin/bash ./install-npm-dependencies
+
+# soci
+COPY ./dependencies/common/install-soci /opt/rstudio-tools/dependencies/common/
+RUN cd /opt/rstudio-tools/dependencies/common && /bin/bash ./install-soci
+
+# yaml-cpp
+COPY ./dependencies/common/install-yaml-cpp /opt/rstudio-tools/dependencies/common/
+RUN cd /opt/rstudio-tools/dependencies/common && /bin/bash ./install-yaml-cpp
+
+# Qt SDK
+COPY ./dependencies/common/install-qt.sh /opt/rstudio-tools/dependencies/linux
+COPY ./dependencies/linux/install-qt-linux /opt/rstudio-tools/dependencies/linux
+RUN export QT_VERSION=5.12.8 && \
+    cd /opt/rstudio-tools/dependencies/linux && /bin/bash ./install-qt-linux
```

Comment:
Since you build with `RSTUDIO_TARGET=Server` the Qt SDK is not used, and you can remove this from the Dockerfile. We should focus our efforts on making this workflow work for Server, and once we've moved to Electron I'll tinker with making that work for Desktop.